XXX-icu-module.md

```diff
@@ -0,0 +1,137 @@
+| Title  | ICU Module                  |
+|--------|-----------------------------|
+| Author | @jasnell                    |
+| Status | REJECTED                    |
+| Date   | 2016-04-20                  |
+
+## Description
+
+The ICU4C library that we use for internationalization contains a significant
+array of additional functionality not currently exposed by the EcmaScript 402
+standard. Some of this additional functionality would be useful to expose via
+a new `'icu'` or (`'uncode'`) module.
+
+## Interface
+
+Initially, the `'icu'` module would provide methods for the following:
+
+1. Character encoding detection. ICU includes code that is able to look at a
+   stream of bytes and apply heuristics to detect the character encoding in
+   use. This is not always an exact match but it does a reasonably good job.
+   We can tune this detection to only look for the character encodings we
+   support in Core (ascii, iso8859-1, utf8 and utf16-le). Two specific APIs
```

Comment:
does icu support win1252? that would be helpful for parsing data from browsers. since the WHATWG has listed that iso8859-1 be a sym for win1252, so it'd be impossible to convert using `'binary'` encoding.